docs/spl-docs/bin-command.md

```diff
@@ -0,0 +1,219 @@
+
+## Syntax
+
+The required syntax is in **bold**.
+
+**bin**\
+[\<bin-options\>...]\
+**\<field\>** [AS \<newfield\>]
+
+### Required Arguments
+
+#### \<field\>
+**Syntax:** `<field>`\
+**Description:** \
+`<field>` specifies the name of the field you wish to bin.
+
+### Optional Arguments
+
+#### \<bin-options\>
+**Syntax:** `<bins> | <minspan> | <span> | <start> | <end> | <aligntime>`\
+**Description:** \
+These are the options for the `bin` command that can be used to create bins for the data in `<field>`.
+
+#### \<newfield\>
+**Syntax:** `AS <string>`\
+**Description:** \
+`<string>` is the name of the new field that will contain the binned data.\
+If not provided, the binned data will overwrite the existing data in the specified `<field>`.
+
+#### \<bins\>
+**Syntax:** `<N>`\
+**Description:** \
+`<N>` is an integer that specifies the maximum number of bins to create.\
+**Default:** 100
+
+#### \<minspan\>
+**Syntax:** `<span-length>`\
+**Description:** \
+Specifies the smallest level of granularity for the bins.
+
+#### \<span\>
+**Syntax:** `<span-length> | <log-span>`\
+**Description:** \
+Specifies the exact size of the bins.
+
+#### \<start\>
+**Syntax:** `<N>`\
+**Description:** \
+`<N>` is a numeric value that signifies the minimum value to be considered when creating bins.\
+This value is disregarded if it exceeds the minimum value present in the data distribution.\
+This option applies only to numerical data and does not apply to time-based data.
+
+#### \<end\>
+**Syntax:** `<N>`\
+**Description:** \
+`<N>` is a numeric value that signifies the maximum value to be considered when creating bins.\
+This value is disregarded if it is less than maximum value present in the data distribution.\
+This option applies only to numerical data and does not apply to time-based data.
+
+#### \<aligntime\>
+**Syntax:** `<absolute-time> | <relative-time> | <N>`\
+**Description:** \
+`<aligntime>` is used to align the bin values for time-based data.\
+The bins created for time data will be aligned to this specific value.\
+`<N>` is a positive integer specifying the time as unix epoch in milliseconds.
+<!-- TODO link earliest and latest pages -->
+Refer to the earliest and latest commands for `<absolute-time>` and `<relative-time>`\
+This option will be ignored if the span has a timescale of `day, week, month or quarter`.\
+By default, all time bins are aligned to UTC epoch 0.
+
+
+#### \<log-span\>
+**Syntax:** `<coefficient>log<base>`\
+**Description:** \
+Sets the span in terms of logarithmic scale.\
+`<coefficient>` is an optional positive numeric value that serves as the coefficient of this logarithmic span.\
+Its value must be less than `<base>` and greater than or equal to `1.0`.\
+**Default Value:** 1.0\
+`<base>` is an optional positive numeric value that serves as the base for the logarithmic span.\
+Its value must be greater than `1.0`.\
+**Default Value:** 10.0
+
```

Comment:
Can you give more detail on how the bin length is calculated when using `log`? I was confused about this (even from Splunk's own documentation) until reading your unit tests https://github.com/siglens/siglens/blob/develop/pkg/segment/aggregations/segaggs_test.go#L2483-L2500

Reply:
addressed in the latest commit